app/mapping/mappers/leadtrustee/LeadTrusteeMapper.scala

```diff
@@ -31,9 +31,16 @@ trait LeadTrusteeMapper[T] extends Mapper[T] {
   def nonUkAddressPage: QuestionPage[NonUkAddress]
 
   def readAddress: Reads[Address] = {
-    (ukAddressYesNoPage.path.read[Boolean] orElse ukCountryOfResidenceYesNoPage.path.read[Boolean]).flatMap {
-      case true => ukAddressPage.path.read[UkAddress].widen[Address]
-      case false => nonUkAddressPage.path.read[Address].widen[Address]
+    lazy val ukAddressYesNo: Reads[Option[Boolean]] = ukAddressYesNoPage.path.readNullable[Boolean]
+    lazy val ukResidencyYesNo: Reads[Option[Boolean]] = ukCountryOfResidenceYesNoPage.path.readNullable[Boolean]
+
+    lazy val ukAddress: Reads[Address] = ukAddressPage.path.read[UkAddress].widen[Address]
+    lazy val nonUkAddress: Reads[Address]  = nonUkAddressPage.path.read[NonUkAddress].widen[Address]
+
+    (ukAddressYesNo orElse ukResidencyYesNo).flatMap {
+      case Some(true) => ukAddress
+      case Some(false) => nonUkAddress
+      case _ => ukAddress orElse nonUkAddress
```

Comment:
I could have replaced this entire method with
`def readAddress: Reads[Address] = {
    lazy val ukAddress: Reads[Address] = ukAddressPage.path.read[UkAddress].widen[Address]
    lazy val nonUkAddress: Reads[Address]  = nonUkAddressPage.path.read[NonUkAddress].widen[Address]

    ukAddress orElse nonUkAddress
    }`